src/ansys/units/quantity.py

```diff
@@ -94,7 +55,15 @@ def __init__(
                 units = copy_from.units
                 value = copy_from.value
 
-        self._value = float(value)
+        if isinstance(value, list):
+            try:
+                import numpy as np
+
+                self._value = np.array(value)
+            except:
+                raise QuantityError.REQUIRES_NUMPY()
```

Comment:
Another approach is to try the import at module scope if it works assign a module variable `_array = np.array` which is otherwise `None`. Obviously, the down side would be that you always try to do the import even if it's unused, but it would makes the code a bit simpler here too.

Reply:
My train of thought for putting the import here is:

- A user can attempt to use a list.
- Receive the `requires_numpy` error
- Install numpy with `pip.main(['install','numpy'])` or similar
- Try the `Quantity` with a list again but it works

Would users ever get invested into a scrip before trying to use a list? I am unsure if that is how a user would interact with this module.

Reply:
No, they need to install numpy so I would add it as an optional package dependency